brave/brave-ios
# New Tab Page & Favicons Refactor

## Summary of Changes

#### Blanket Refactor Issues
This pull request fixes #2578
This pull request fixes #2579 

#### NTP Issues
This pull request fixes #2380
This pull request fixes #2071
This pull request fixes https://github.com/brave/brave-ios/issues/2036
This pull request fixes https://github.com/brave/brave-ios/issues/2351
This pull request fixes https://github.com/brave/brave-ios/issues/2059
This pull request fixes https://github.com/brave/brave-ios/issues/2516

#### Keyboard Issues
*Note that the "Auto-Open Keyboard" preference has been removed*

This pull request fixes https://github.com/brave/brave-ios/issues/2078
This pull request fixes https://github.com/brave/brave-ios/issues/2073
This pull request fixes https://github.com/brave/brave-ios/issues/2321 (because the pref no longer exists)

#### Favicon Issues
This pull request fixes https://github.com/brave/brave-ios/issues/2051
This pull request fixes https://github.com/brave/brave-ios/issues/1162
This pull request fixes https://github.com/brave/brave-ios/issues/933
This pull request fixes https://github.com/brave/brave-ios/issues/528
This pull request fixes https://github.com/brave/brave-ios/issues/2362
This pull request fixes https://github.com/brave/brave-ios/issues/2485
This pull request fixes https://github.com/brave/brave-ios/issues/2099
This pull request fixes https://github.com/brave/brave-ios/issues/1865
This pull request fixes https://github.com/brave/brave-ios/issues/1030
This pull request fixes https://github.com/brave/brave-ios/issues/2364

## Submitter Checklist:


## Test Plan:

- Test New Tab Page related functions:
   - Verify stats appear & function
   - Verify favorites appear and are left-aligned even with less than 4 items
   - Verify show more only appears when there's more than the visible number of favorites on NTP (between 4-6 depending on iPhone vs iPad portrait & landscape)
   - Verify iPhone/iPad landscape vs portrait
   - Verify background images
   - Verify background images source button
   - Verify SI's & SI button
   - Verify CHP's & QR code button
   - Verify private mode callouts
   - Verify themes
   - Verify no-image-bg light theme shows dark-blue bg now and no gradient

- Test all places favicons are seen: Favorites, Bookmarks, Add Bookmark, History, Search, Back/Forward list, Tab Tray, Rewards Panel, Rewards Tipping/Auto-Contribute/Pending Contributions list, etc.
    - Verify custom icons are shown for super-referrers 
    - Verify bundled icons are shown until you visit the page (therefore parsing and downloading the websites up-to-date favicon)
    - Verify websites that only have small favicons display monograms on the favorites overlay & NTP. (i.e. https://eqbank.ca) but display the icon in other places
    - Verify websites that don't have any valid favicon display monograms (e.g. https://kylehickinson.com)

- Test new favorites overlay basic functionality (tap, edit, reorder, scroll, etc.) 

- Test upgrading from older versions that already have favicons downloaded

## Screenshots:

<details>
<summary>Screenshots</summary>

![Simulator Screen Shot - iPhone 11 Pro - 2020-05-25 at 13 41 06](https://user-images.githubusercontent.com/529104/82833912-6b89ed00-9e8d-11ea-9e6f-62200d45867d.png)

![Simulator Screen Shot - iPhone 11 Pro - 2020-05-25 at 13 41 35](https://user-images.githubusercontent.com/529104/82833919-747abe80-9e8d-11ea-8dc6-14243efc84d8.png)

![Simulator Screen Shot - iPhone 11 Pro - 2020-05-25 at 13 44 15](https://user-images.githubusercontent.com/529104/82834024-d4716500-9e8d-11ea-89ae-c853b7b008ed.png)

![Simulator Screen Shot - iPhone 11 Pro - 2020-05-25 at 13 45 17](https://user-images.githubusercontent.com/529104/82834063-f9fe6e80-9e8d-11ea-8354-01b131057246.png)

</details>


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
